### ISSUE TITLE
[Bug] [autoscaler] ray up setup command output doesn't show (at least on GCP)

### ISSUE THREAD

**Issue by DmitriGekhtman** [ray-project/ray#18774]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Clusters

### What happened + What you expected to happen

Related to command output issues described here:
https://github.com/ray-project/ray/issues/16539
https://github.com/ray-project/ray/pull/18653

I will investigate and update.

### Reproduction script

Not yet, just reminding myself to look at this.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by ijrsvt** [comment]
@DmitriGekhtman I don't know exactly what is going on here, but I suspect it is something with buffering STDOUT/STDERR (and it not getting sent over when the command fails).

One weird thing I've noticed is that explicitly writing a setup command that prints to STDOUT/STDERR and then exiting (i.e. `>&2 echo "error"; exit 1` usually works).

**Comment by DmitriGekhtman** [comment]
I don't quite see which part of the error handling here is supposed to print error output.
https://github.com/ray-project/ray/blob/337005d5a544c9fff4be2929e2c30bfc02d2bbd3/python/ray/autoscaler/_private/command_runner.py#L435-L451

**Comment by DmitriGekhtman** [comment]
I guess we're expecting `is_output_redirected()` to be True and if it's True we expect the subprocess call to write to stdout/stderr.

**Comment by DmitriGekhtman** [comment]
Hmm, no, `is_output_redirected()` returns False by default. 
https://github.com/ray-project/ray/blob/337005d5a544c9fff4be2929e2c30bfc02d2bbd3/python/ray/autoscaler/_private/commands.py#L174

**Comment by DmitriGekhtman** [comment]
@richardliaw it's probably been a while, but do you remember by what means CommandRunner.run is supposed to produce error output for failed commands?

**Comment by DmitriGekhtman** [comment]
Ok, yeah, if you just put in a faulty setup command "blah" you do get `bash: blah: command not found`

**Comment by DmitriGekhtman** [comment]
Hmm, instead of trying to figure out where the error is getting eaten, I'll try to propagate the error to the block of red text at the end of the output.
Might be slightly redundant in most cases.

**Comment by DmitriGekhtman** [comment]
Will probably have to come back to this next week, but noting to self -- the fix/workaround is to extract the subprocess error and display it with the current `fail_msg`.

**Comment by richardliaw** [comment]
Interesting, wonder why this doesn’t show up for AWS

**Comment by ericl** [comment]
I'm downgrading this to P1 since it is not meeting the priority of a P0 regression (P0s are interrupts that take priority over other issues, clearly this isn't the case if work is deferred).

**Comment by msaroufim** [comment]
Hi folks I came in from this issue https://github.com/ray-project/ray/issues/18755 and wanted to confirm that I'm seeing this for AWS

Repro steps is https://docs.ray.io/en/latest/cluster/quickstart.html#ref-cluster-quick-start until `ray up -y config.yaml`

A machine was indeed created in my console but I kept getting this error even though I've enabled all ingoing and outgoing traffic. I also confirmed that aws credentials were correctlly stored in `.aws/credentials`

Also tested on my local macbook same issue

```

ubuntu@ip-172-31-12-140:~$ ray up -y config.yaml
Cluster: minimal

2021-10-08 23:57:32,857 INFO util.py:282 -- setting max workers for head node type to 0
2021-10-08 23:57:32,857 INFO util.py:286 -- setting max workers for ray.worker.default to 2
Loaded cached provider configuration
If you experience issues with the cloud provider, try re-running the command with --no-config-cache.
AWS config
  IAM Profile: ray-autoscaler-v1 [default]
  EC2 Key pair (all available node types): ray-autoscaler_2_us-west-2 [default]
  VPC Subnets (all available node types): subnet-00b909d3664ca1927, subnet-0f73c4179776b1b44, subnet-0954508b8c04c4190 [default]
  EC2 Security groups (all available node types): sg-0115e4342d6d47c84 [default]
  EC2 AMI (all available node types): ami-0a2363a9cff180a64

Updating cluster configuration and running full setup.
Cluster Ray runtime will be restarted. Confirm [y/N]: y [automatic, due to --yes]

<1/1> Setting up head node
  Prepared bootstrap config
  New status: waiting-for-ssh
  [1/7] Waiting for SSH to become available
    Running `uptime` as a test.
    Fetched IP: 34.209.103.80
ssh: connect to host 34.209.103.80 port 22: Connection timed out
    SSH still not available (SSH command failed.), retrying in 5 seconds.
ssh: connect to host 34.209.103.80 port 22: Connection timed out
    SSH still not available (SSH command failed.), retrying in 5 seconds.
ssh: connect to host 34.209.103.80 port 22: Connection timed out
```

**Comment by richardliaw** [comment]
Hey @msaroufim I think this is a different issue, let me make a new thread with your report.
